src/main.rs

```diff
@@ -81,12 +81,28 @@ fn run() -> Result<ExitCode> {
     }
 
     ensure_search_pattern_is_not_a_path(&opts)?;
-    let pattern_regex = build_pattern_regex(&opts)?;
+    let pattern = &opts.pattern;
+    let exprs = &opts.exprs;
+    let empty = Vec::new();
+
+    let pattern_regexps = exprs
+        .as_ref()
+        .unwrap_or(&empty)
+        .iter()
+        .chain([pattern])
+        .map(|pat| build_pattern_regex(pat, &opts))
+        .collect::<Result<Vec<String>>>()?;
+
+    let config = construct_config(opts, &pattern_regexps)?;
 
-    let config = construct_config(opts, &pattern_regex)?;
-    ensure_use_hidden_option_for_leading_dot_pattern(&config, &pattern_regex)?;
-    let re = build_regex(pattern_regex, &config)?;
-    walk::scan(&search_paths, Arc::new(re), Arc::new(config))
+    ensure_use_hidden_option_for_leading_dot_pattern(&config, &pattern_regexps)?;
+
+    let regexps = pattern_regexps
+        .iter()
+        .map(|pat| build_regex(pat.to_string(), &config))
```

Comment:
isn't `pat` a string already?

Reply:
There is an error message when I remove the `to_string()`:

```
error[E0308]: mismatched types: expected struct `std::string::String`, found `&std::string::String
```

Reply:
Then I guess you could use `.into_iter()` instead of `.iter()` (and remove `.to_string()`)?

Reply:
https://github.com/sharkdp/fd/compare/d8196a109e81e812ac9e558363a210ca2d660e11..1409ad6d306b400c07d75538a804810d8a569bc0